Packet++/header/TcpReassembly.h

```diff
@@ -308,6 +359,13 @@ class TcpReassembly
 	 */
 	int isConnectionOpen(const ConnectionData& connection) const;
 
+	/**
+	 * Clean up the closed connections from the memory
+	 * @param[in] currentTime Current time
+	 * @param[in] maxNumToClean The maximum number of items to be cleaned up per one call. This parameter, when its value is not zero, overrides the value that was set by the constructor.
+	 */
+	void purgeClosedConnections(time_t currentTime = time(NULL), uint32_t maxNumToClean = 0);

```

Comment:
do you really need the param `currentTime`? It is always set to `time(NULL)`

Reply:
Param has been removed.

Reply:
This parameter allows to use the value that is got earlier and saved in local/global variable. It's a small optimization. For example: I have  a global variable of type `volatile time_t` in my code. This variable is being updated by certain thread. Therefore I just use this variable instead of calling the function `time`.

Reply:
I am planning to change the signature of this method as:

`uint32_t purgeClosedConnections(time_t currentTime = time(NULL), uint32_t maxNumToClean = 0);`

The return value can be useful for performance measurement.

Reply:
Following code is used in `reassemblyPacket`:

```
  // automatic cleanup
  if(m_RemoveConnInfo == true)
  {
    time_t currentTime = time(NULL);
    if(currentTime >= m_PurgeTimepoint)
    {
      purgeClosedConnections(currentTime, 0);
      m_PurgeTimepoint = time(NULL) + PURGE_FREQ_SECS;
    }
  }
```
Without this parameter time() will be called twice.

Reply:
I do understand the optimization, however we should consider what is the overhead of `time()` (which is something I'm not sure about). If the overhead is small then it's probably better and cleaner to call it one more time. If the overhead is big then I agree with your approach.

Please let me know what you think

Reply:
The overhead is small. I will remove this parameter.